src/main/java/com/bookingApp/bookingApp/controller/ReservationController.java

```diff
@@ -0,0 +1,51 @@
+package com.bookingApp.bookingApp.controller;
+
+import com.bookingApp.bookingApp.DTOs.ReservationBackDto;
+import com.bookingApp.bookingApp.DTOs.ReservationDto;
+import com.bookingApp.bookingApp.DTOs.ScreeningInfoDto;
+import com.bookingApp.bookingApp.DTOs.ScreeningsDto;
+import com.bookingApp.bookingApp.service.IReservationService;
+import com.bookingApp.bookingApp.service.ReservationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/screenings", produces = "application/json")
+public class ReservationController {
+    IReservationService reservationService;
+
+    @Autowired
+    public ReservationController(ReservationService service){
+        this.reservationService = service;
```

Comment:
dlaczego używasz interfejsu jako pola a klasy konkretnej jako parametru? to łamie pewne reguły?

Reply:
faktycznie tutaj blad